src/docfx/lib/PathUtil.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Docs
+{
+    /// <summary>
+    /// Provide utils of path
+    /// </summary>
+    public static class PathUtil
+    {
+        /// <summary>
+        /// Create a relative path from one path to another file.
+        /// Use this over <see cref="Path.GetRelativePath(string, string)"/> when
+        /// <paramref name="fileRelativeTo"/> is a file.
+        /// </summary>
+        /// <param name="fileRelativeTo">The file path relative to</param>
+        /// <param name="path">The original path </param>
+        /// <returns>The relative path</returns>
+        public static string GetRelativePathToFile(string fileRelativeTo, string path)
+        {
+            var directory = Path.GetDirectoryName(fileRelativeTo);
+            if (string.IsNullOrEmpty(directory))
+            {
+                return path;
+            }
+
+            return Path.GetRelativePath(directory, path);
+        }
+
+        /// <summary>
+        /// A normalized folder always ends with `/`, does not contain `\` and does not have consegative `.` or `/`.
+        /// </summary>
+        /// <param name="path">The folder path want to be normalized</param>
+        /// <returns>The normalized folder path</returns>
+        public static string NormalizeFolder(string path)
+        {
+            Debug.Assert(path.IndexOfAny(Path.GetInvalidPathChars()) < 0);
+
+            var str = Normalize(path);
+            if (str.Length == 0 || str == "/")
+            {
+                return "./";
+            }
+
+            if (!str.EndsWith('/'))
+            {
+                str += '/';
+            }
+            return str;
+        }
+
+        /// <summary>
+        /// A normalized file cannot end with `/`, does not contain `\` and does not have consegative `.` or `/`.
+        /// </summary>
+        /// <param name="path">The file path want to be normalized</param>
+        /// <returns>The normalized file path</returns>
+        public static string NormalizeFile(string path)
+        {
+            Debug.Assert(Path.GetDirectoryName(path).IndexOfAny(Path.GetInvalidPathChars()) < 0);
+            Debug.Assert(Path.GetFileName(path).IndexOfAny(Path.GetInvalidFileNameChars()) < 0);
+
+            return Normalize(path);
+        }
+
+        private static string Normalize(string path)
+        {
+            path = path.Replace('\\', '/');
+
+            var needReorder = false;
+            if (path.IndexOf('.') != -1)
+            {
+                needReorder = true;
+            }
+
+            if (!needReorder)
+            {
+                return path;
+            }
+
+            var parentCount = 0;
+            var rooted = path[0] == '/';
```

Comment:
>var rooted = path[0] == '/'; [](start = 12, length = 28)

what about assert the path is not rooted? Now `/../a` will be normalized to `/a`